# Подготовка файлов и программ

Некоторые программы ставятся дважды из разных источников для демонстрации различных вариантов в случае неполадок. Для оптимизации времени работы необходимо удалить лишний код.

In [1]:
pip install -q condacolab

In [2]:
import condacolab

In [3]:
condacolab.install()

✨🍰✨ Everything looks OK!


In [4]:
!conda install -c bioconda seqtk

Solving environment: - \ | / - \ | / done

## Package Plan ##

  environment location: /usr/local

  added / updated specs:
    - seqtk


The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    ca-certificates-2021.10.8  |       ha878542_0         139 KB  conda-forge
    certifi-2021.10.8          |   py37h89c1867_2         145 KB  conda-forge
    conda-4.12.0               |   py37h89c1867_0         1.0 MB  conda-forge
    seqtk-1.3                  |       h5bf99c6_3          38 KB  bioconda
    ------------------------------------------------------------
                                           Total:         1.3 MB

The following NEW packages will be INSTALLED:

  seqtk              bioconda/linux-64::seqtk-1.3-h5bf99c6_3

The following packages will be UPDATED:

  ca-certificates                      2020.12.5-ha878542_0 --> 2021.10.8-ha878542_0
  certifi                  

In [5]:
!conda install -c bioconda trimmomatic

Solving environment: | / - \ | / - \ | / - \ | done

## Package Plan ##

  environment location: /usr/local

  added / updated specs:
    - trimmomatic


The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    openjdk-8.0.312            |       h7f98852_0        97.6 MB  conda-forge
    python_abi-3.7             |          2_cp37m           4 KB  conda-forge
    trimmomatic-0.39           |       hdfd78af_2         144 KB  bioconda
    ------------------------------------------------------------
                                           Total:        97.8 MB

The following NEW packages will be INSTALLED:

  openjdk            conda-forge/linux-64::openjdk-8.0.312-h7f98852_0
  trimmomatic        bioconda/noarch::trimmomatic-0.39-hdfd78af_2

The following packages will be UPDATED:

  python_abi                                    3.7-1_cp37m --> 3.7-2_cp37m



python_abi

In [6]:
!pip install macs2 intervene

     |████████████████████████████████| 2.1 MB 5.2 MB/s 
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
    Preparing wheel metadata ... done
     |████████████████████████████████| 5.5 MB 30.9 MB/s 
     |████████████████████████████████| 12.5 MB 22.2 MB/s 
     |████████████████████████████████| 11.2 MB 41.9 MB/s 
     |████████████████████████████████| 11.3 MB 32.9 MB/s 
  Using cached numpy-1.21.6-cp37-cp37m-manylinux_2_12_x86_64.manylinux2010_x86_64.whl (15.7 MB)
     |████████████████████████████████| 38.1 MB 1.4 MB/s 
     |████████████████████████████████| 292 kB 33.9 MB/s 
     |████████████████████████████████| 1.1 MB 41.7 MB/s 
     |████████████████████████████████| 40 kB 4.0 MB/s 
     |████████████████████████████████| 247 kB 3.9 MB/s 
     |████████████████████████████████| 98 kB 7.0 MB/s 
     |████████████████████████████████| 4.3 MB 28.0 MB/s 
     |████████████████████████████████| 930 kB 38.2 MB/s 
     |█████████████████████

In [1]:
!apt-get install bedtools

Reading package lists... Done
Building dependency tree       
Reading state information... Done
The following packages were automatically installed and are no longer required:
  libnvidia-common-460 nsight-compute-2020.2.0
Use 'apt autoremove' to remove them.
The following NEW packages will be installed:
  bedtools
0 upgraded, 1 newly installed, 0 to remove and 42 not upgraded.
Need to get 577 kB of archives.
After this operation, 2,040 kB of additional disk space will be used.
Get:1 http://archive.ubuntu.com/ubuntu bionic/universe amd64 bedtools amd64 2.26.0+dfsg-5 [577 kB]
Fetched 577 kB in 1s (721 kB/s)
Selecting previously unselected package bedtools.
(Reading database ... 155203 files and directories currently installed.)
Preparing to unpack .../bedtools_2.26.0+dfsg-5_amd64.deb ...
Unpacking bedtools (2.26.0+dfsg-5) ...
Setting up bedtools (2.26.0+dfsg-5) ...


In [2]:
!apt-get install bowtie2 samtools

Reading package lists... Done
Building dependency tree       
Reading state information... Done
The following packages were automatically installed and are no longer required:
  libnvidia-common-460 nsight-compute-2020.2.0
Use 'apt autoremove' to remove them.
The following additional packages will be installed:
  cwltool libc-ares2 libhts2 libjs-bootstrap nodejs nodejs-doc
  python-asn1crypto python-avro python-cachecontrol python-certifi
  python-cffi-backend python-chardet python-cryptography python-enum34
  python-html5lib python-idna python-ipaddress python-isodate python-lockfile
  python-mistune python-openssl python-pkg-resources python-pyparsing
  python-rdflib python-rdflib-jsonld python-requests python-ruamel.yaml
  python-schema-salad python-shellescape python-six python-sparqlwrapper
  python-typing python-urllib3 python-webencodings
Suggested packages:
  bowtie2-examples python-cryptography-doc python-cryptography-vectors
  python-enum34-doc python-genshi python-lxml pytho

In [3]:
!conda install -c bioconda bowtie2

Solving environment: | / - \ | / - \ | / - \ done

## Package Plan ##

  environment location: /usr/local

  added / updated specs:
    - bowtie2


The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    bowtie2-2.3.5.1            |   py37he513fc3_0        12.0 MB  bioconda
    perl-5.32.1                | 0_h7f98852_perl5        14.5 MB  conda-forge
    tbb-2020.2                 |       h4bd325d_4         1.5 MB  conda-forge
    ------------------------------------------------------------
                                           Total:        27.9 MB

The following NEW packages will be INSTALLED:

  bowtie2            bioconda/linux-64::bowtie2-2.3.5.1-py37he513fc3_0
  perl               conda-forge/linux-64::perl-5.32.1-0_h7f98852_perl5
  tbb                conda-forge/linux-64::tbb-2020.2-h4bd325d_4



perl-5.32.1          | 14.5 MB   | : 100% 1.0/1 [00:04<00:00,  4

## Установка FastQC

In [4]:
!java -version

openjdk version "1.8.0_312"
OpenJDK Runtime Environment (Zulu 8.58.0.13-CA-linux64) (build 1.8.0_312-b07)
OpenJDK 64-Bit Server VM (Zulu 8.58.0.13-CA-linux64) (build 25.312-b07, mixed mode)


In [5]:
!wget https://www.bioinformatics.babraham.ac.uk/projects/fastqc/fastqc_v0.11.9.zip

--2022-03-07 17:59:21--  https://www.bioinformatics.babraham.ac.uk/projects/fastqc/fastqc_v0.11.9.zip
Resolving www.bioinformatics.babraham.ac.uk (www.bioinformatics.babraham.ac.uk)... 149.155.133.4
Connecting to www.bioinformatics.babraham.ac.uk (www.bioinformatics.babraham.ac.uk)|149.155.133.4|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 10249221 (9.8M) [application/zip]
Saving to: ‘fastqc_v0.11.9.zip’

fastqc_v0.11.9.zip  100%[===================>]   9.77M  3.90MB/s    in 2.5s    

2022-03-07 17:59:24 (3.90 MB/s) - ‘fastqc_v0.11.9.zip’ saved [10249221/10249221]



In [6]:
!unzip fastqc_v0.11.9.zip

Archive:  fastqc_v0.11.9.zip
  inflating: FastQC/cisd-jhdf5.jar   
   creating: FastQC/Configuration/
  inflating: FastQC/Configuration/adapter_list.txt  
  inflating: FastQC/Configuration/contaminant_list.txt  
  inflating: FastQC/Configuration/limits.txt  
  inflating: FastQC/fastqc           
  inflating: FastQC/fastqc_icon.ico  
   creating: FastQC/Help/
   creating: FastQC/Help/1 Introduction/
   creating: FastQC/Help/1 Introduction/.svn/
  inflating: FastQC/Help/1 Introduction/.svn/entries  
   creating: FastQC/Help/1 Introduction/.svn/props/
   creating: FastQC/Help/1 Introduction/.svn/text-base/
  inflating: FastQC/Help/1 Introduction/.svn/text-base/1.1 What is FastQC.html.svn-base  
   creating: FastQC/Help/1 Introduction/.svn/tmp/
   creating: FastQC/Help/1 Introduction/.svn/tmp/props/
  inflating: FastQC/Help/1 Introduction/1.1 What is FastQC.html  
   creating: FastQC/Help/2 Basic Operations/
   creating: FastQC/Help/2 Basic Operations/.svn/
  inflating: FastQC/Help/2 Basic

In [7]:
!chmod a+x FastQC/fastqc

In [8]:
!./FastQC/fastqc --help


            FastQC - A high throughput sequence QC analysis tool

SYNOPSIS

	fastqc seqfile1 seqfile2 .. seqfileN

    fastqc [-o output dir] [--(no)extract] [-f fastq|bam|sam] 
           [-c contaminant file] seqfile1 .. seqfileN

DESCRIPTION

    FastQC reads a set of sequence files and produces from each one a quality
    control report consisting of a number of different modules, each one of 
    which will help to identify a different potential type of problem in your
    data.
    
    If no files to process are specified on the command line then the program
    will start as an interactive graphical application.  If files are provided
    on the command line then the program will run with no user interaction
    required.  In this mode it is suitable for inclusion into a standardised
    analysis pipeline.
    
    The options for the program as as follows:
    
    -h --help       Print this help file and exit
    
    -v --version    Print the version of the program and exit

# Выравнивание

## Скачиваем чтения

Для примера чтения были взяты из эксперимента https://www.encodeproject.org/experiments/ENCSR000AKC/

In [9]:
# ChIP-seq на гистоновой метке
!wget https://www.encodeproject.org/files/ENCFF738ERP/@@download/ENCFF738ERP.fastq.gz
!wget https://www.encodeproject.org/files/ENCFF134NCG/@@download/ENCFF134NCG.fastq.gz
# ChIP-seq контроль
!wget https://www.encodeproject.org/files/ENCFF222VRH/@@download/ENCFF222VRH.fastq.gz

--2022-03-07 17:59:45--  https://www.encodeproject.org/files/ENCFF738ERP/@@download/ENCFF738ERP.fastq.gz
Resolving www.encodeproject.org (www.encodeproject.org)... 34.211.244.144
Connecting to www.encodeproject.org (www.encodeproject.org)|34.211.244.144|:443... connected.
HTTP request sent, awaiting response... 307 Temporary Redirect
Location: https://encode-public.s3.amazonaws.com/2016/10/07/2fcb2fac-f190-4e22-b1f6-bf48983c9666/ENCFF738ERP.fastq.gz?response-content-disposition=attachment%3B%20filename%3DENCFF738ERP.fastq.gz&AWSAccessKeyId=ASIATGZNGCNX5WD2ZKUR&Signature=%2BIZnynTD8s3FOR%2FyBPGa%2FFSiDJw%3D&x-amz-security-token=IQoJb3JpZ2luX2VjEEIaCXVzLXdlc3QtMiJHMEUCIAbvj65vd86oXCN2OI%2FSa2zWVLo4xzssXBQJshq7oLeYAiEAt0zJumE%2BSaQeUp5uQcanqBeWIb2GrOC0dEdxQY1V1LUq0gQIGxAAGgwyMjA3NDg3MTQ4NjMiDAel%2FLrLuPTq3ZS4PSqvBCbljA55zEMpTdBH7gJzBZhMITvv3XsatAQpXDsCUryUbAFrLZRtteBBlxYe8d01CHKkx5kyJ%2BL%2FgjpoIFXDk44ugrYgvEnoGJ3wZJ%2FrotZOz7cP8SvZmBXss8U5XHd6gt8rAg49gu3BQgYU3PlRPdcnJZ%2BVj18gQ1GKfVKMzvH

In [10]:
# распаковка ахрива
!gzip -d /content/ENCFF738ERP.fastq.gz /content/ENCFF134NCG.fastq.gz /content/ENCFF222VRH.fastq.gz

## FastQC (или multiQC может быть использован в работе)

In [11]:
!./FastQC/fastqc /content/ENCFF738ERP.fastq
!./FastQC/fastqc /content/ENCFF134NCG.fastq
!./FastQC/fastqc /content/ENCFF222VRH.fastq

Started analysis of ENCFF738ERP.fastq
Approx 5% complete for ENCFF738ERP.fastq
Approx 10% complete for ENCFF738ERP.fastq
Approx 15% complete for ENCFF738ERP.fastq
Approx 20% complete for ENCFF738ERP.fastq
Approx 25% complete for ENCFF738ERP.fastq
Approx 30% complete for ENCFF738ERP.fastq
Approx 35% complete for ENCFF738ERP.fastq
Approx 40% complete for ENCFF738ERP.fastq
Approx 45% complete for ENCFF738ERP.fastq
Approx 50% complete for ENCFF738ERP.fastq
Approx 55% complete for ENCFF738ERP.fastq
Approx 60% complete for ENCFF738ERP.fastq
Approx 65% complete for ENCFF738ERP.fastq
Approx 70% complete for ENCFF738ERP.fastq
Approx 75% complete for ENCFF738ERP.fastq
Approx 80% complete for ENCFF738ERP.fastq
Approx 85% complete for ENCFF738ERP.fastq
Approx 90% complete for ENCFF738ERP.fastq
Approx 95% complete for ENCFF738ERP.fastq
Analysis complete for ENCFF738ERP.fastq
Started analysis of ENCFF134NCG.fastq
Approx 5% complete for ENCFF134NCG.fastq
Approx 10% complete for ENCFF134NCG.fastq
Appr

In [12]:
!conda install -c bioconda -c conda-forge multiqc

Solving environment: / - \ | / - \ | / - \ | / - \ | / - \ | / - \ | / - \ | / - \ | / - \ | / - \ | / - \ | / - \ | / - \ | / - \ | / - \ | / - \ | / - \ | / - \ | / - \ | / - \ | / done

## Package Plan ##

  environment location: /usr/local

  added / updated specs:
    - multiqc


The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    click-8.1.3                |   py37h89c1867_0         145 KB  conda-forge
    colorama-0.4.4             |     pyh9f0ad1d_0          18 KB  conda-forge
    coloredlogs-15.0.1         |     pyhd8ed1ab_3          40 KB  conda-forge
    colormath-3.0.0            |             py_2          34 KB  conda-forge
    commonmark-0.9.1           |             py_0          46 KB  conda-forge
    cycler-0.11.0 

In [13]:
!multiqc .


  /// ]8;id=229632;https://multiqc.info\MultiQC]8;;\ 🔍 | v1.12

|           multiqc | Search path : /content
|         searching | ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 100% 286/286  
|            fastqc | Found 3 reports
|           multiqc | Compressing plot data
|           multiqc | Report      : multiqc_report.html
|           multiqc | Data        : multiqc_data
|           multiqc | MultiQC complete


Далее необходимо проанализировать выдачу fastqc, произвести фильтрацию или подрезание чтений в случае необходимости. Для примера указано использование trimmomatic, также можно использовать seqtk.

Высокий уровень дедуплицирования ридов в ENCFF222VRH. Проведем подрезание

In [14]:
!trimmomatic SE -phred33 ENCFF222VRH.fastq ENCFF222VRH_trimmed.fastq ILLUMINACLIP:TruSeq3-SE:2:30:10 \
  LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 MINLEN:36

TrimmomaticSE: Started with arguments:
 -phred33 ENCFF222VRH.fastq ENCFF222VRH_trimmed.fastq ILLUMINACLIP:TruSeq3-SE:2:30:10 LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 MINLEN:36
Automatically using 2 threads
java.io.FileNotFoundException: /content/TruSeq3-SE (No such file or directory)
	at java.io.FileInputStream.open0(Native Method)
	at java.io.FileInputStream.open(FileInputStream.java:195)
	at java.io.FileInputStream.<init>(FileInputStream.java:138)
	at org.usadellab.trimmomatic.fasta.FastaParser.parse(FastaParser.java:54)
	at org.usadellab.trimmomatic.trim.IlluminaClippingTrimmer.loadSequences(IlluminaClippingTrimmer.java:110)
	at org.usadellab.trimmomatic.trim.IlluminaClippingTrimmer.makeIlluminaClippingTrimmer(IlluminaClippingTrimmer.java:71)
	at org.usadellab.trimmomatic.trim.TrimmerFactory.makeTrimmer(TrimmerFactory.java:32)
	at org.usadellab.trimmomatic.Trimmomatic.createTrimmers(Trimmomatic.java:59)
	at org.usadellab.trimmomatic.TrimmomaticSE.run(TrimmomaticSE.java:318)
	at org.u

In [15]:
!./FastQC/fastqc ENCFF222VRH_trimmed.fastq

Started analysis of ENCFF222VRH_trimmed.fastq
Approx 5% complete for ENCFF222VRH_trimmed.fastq
Approx 10% complete for ENCFF222VRH_trimmed.fastq
Approx 15% complete for ENCFF222VRH_trimmed.fastq
Approx 20% complete for ENCFF222VRH_trimmed.fastq
Approx 25% complete for ENCFF222VRH_trimmed.fastq
Approx 30% complete for ENCFF222VRH_trimmed.fastq
Approx 35% complete for ENCFF222VRH_trimmed.fastq
Approx 40% complete for ENCFF222VRH_trimmed.fastq
Approx 45% complete for ENCFF222VRH_trimmed.fastq
Approx 50% complete for ENCFF222VRH_trimmed.fastq
Approx 55% complete for ENCFF222VRH_trimmed.fastq
Approx 60% complete for ENCFF222VRH_trimmed.fastq
Approx 65% complete for ENCFF222VRH_trimmed.fastq
Approx 70% complete for ENCFF222VRH_trimmed.fastq
Approx 75% complete for ENCFF222VRH_trimmed.fastq
Approx 80% complete for ENCFF222VRH_trimmed.fastq
Approx 85% complete for ENCFF222VRH_trimmed.fastq
Approx 90% complete for ENCFF222VRH_trimmed.fastq
Approx 95% complete for ENCFF222VRH_trimmed.fastq
Analy

## Выравнивание на хромосому

Следует выбрать одну хромосому, потому что ресурсы google colab ограничены. В случае проблем, таких как Segmentation fault, error 139, следует выбрать хромосому меньшего размера.

Последовательность нуклеотидов хромосомы можно скачать по адресу https://hgdownload.soe.ucsc.edu/goldenPath/hg38/chromosomes/

Взял 13 хромосому, так как не хватало места в colab

In [ ]:
!wget https://hgdownload.soe.ucsc.edu/goldenPath/hg38/chromosomes/chr13.fa.gz
!gzip -d chr13.fa.gz

--2022-03-07 21:13:32--  https://hgdownload.soe.ucsc.edu/goldenPath/hg38/chromosomes/chr13.fa.gz
Resolving hgdownload.soe.ucsc.edu (hgdownload.soe.ucsc.edu)... 128.114.119.163
Connecting to hgdownload.soe.ucsc.edu (hgdownload.soe.ucsc.edu)|128.114.119.163|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 31870393 (30M) [application/x-gzip]
Saving to: ‘chr13.fa.gz’

chr13.fa.gz         100%[===================>]  30.39M  11.0MB/s    in 2.8s    

2022-03-07 21:13:36 (11.0 MB/s) - ‘chr13.fa.gz’ saved [31870393/31870393]



In [ ]:
%%time
!bowtie2-build chr13.fa chromosome_index

Settings:
  Output files: "chromosome_index.*.bt2"
  Line rate: 6 (line is 64 bytes)
  Lines per side: 1 (side is 64 bytes)
  Offset rate: 4 (one in 16)
  FTable chars: 10
  Strings: unpacked
  Max bucket size: default
  Max bucket size, sqrt multiplier: default
  Max bucket size, len divisor: 4
  Difference-cover sample period: 1024
  Endianness: little
  Actual local endianness: little
  Sanity checking: disabled
  Assertions: disabled
  Random seed: 0
  Sizeofs: void*:8, int:4, long:8, size_t:8
Input files DNA, FASTA:
  chr13.fa
Building a SMALL index
Reading reference sizes
  Time reading reference sizes: 00:00:02
Calculating joined length
Writing header
Reserving space for joined string
Joining reference sequences
  Time to join reference sequences: 00:00:00
bmax according to bmaxDivN setting: 24495781
Using parameters --bmax 18371836 --dcv 1024
  Doing ahead-of-time memory usage test
  Passed!  Constructing with these parameters: --bmax 18371836 --dcv 1024
Constructing suffix-arr

In [ ]:
!mkdir bowtie2_res

In [ ]:
%%time
!bowtie2  -p 2 \
          -x chromosome_index \
          -U ENCFF222VRH_trimmed.fastq \
          -S bowtie2_res/res_VRH.sam

54128905 reads; of these:
  54128905 (100.00%) were unpaired; of these:
    46986193 (86.80%) aligned 0 times
    2063611 (3.81%) aligned exactly 1 time
    5079101 (9.38%) aligned >1 times
13.20% overall alignment rate
CPU times: user 15.3 s, sys: 2.12 s, total: 17.5 s
Wall time: 53min 41s


In [ ]:
%%time
!bowtie2  -p 2 \
          -x chromosome_index \
          -U ENCFF134NCG.fastq \
          -S bowtie2_res/res_NCG.sam

22026319 reads; of these:
  22026319 (100.00%) were unpaired; of these:
    19272245 (87.50%) aligned 0 times
    783224 (3.56%) aligned exactly 1 time
    1970850 (8.95%) aligned >1 times
12.50% overall alignment rate
CPU times: user 5.96 s, sys: 782 ms, total: 6.75 s
Wall time: 20min 41s


In [ ]:
%%time
!bowtie2  -p 2 \
          -x chromosome_index \
          -U ENCFF738ERP.fastq \
          -S bowtie2_res/res_ERP.sam

20301162 reads; of these:
  20301162 (100.00%) were unpaired; of these:
    17763189 (87.50%) aligned 0 times
    724390 (3.57%) aligned exactly 1 time
    1813583 (8.93%) aligned >1 times
12.50% overall alignment rate
CPU times: user 5.52 s, sys: 725 ms, total: 6.25 s
Wall time: 19min 4s


Проанализируйте выдачу bowtie. Почему процент выравниваний получился именно таким?

Имеет смысл для дальнейшего анализа отобрать уникально картированные риды.

Преобразование .sam в .bam (не требуется если запускать macs2 на .sam файлах)

In [ ]:
!samtools view -h -S -b -o bowtie2_res/res_ASU.bam bowtie2_res/res_NCG.sam
!samtools view -h -S -b -o bowtie2_res/res_ARK.bam bowtie2_res/res_ERP.sam

На этом моменте можно скачать себе на компьютер все полученные .bam или .sam файлы на случай сбоя.

## Peak calling

Помимо macs2 может быть использован пакет Homer для поиска пиков, а также macs3.

In [ ]:
!mkdir macs2
!mkdir macs22

mkdir: cannot create directory ‘macs2’: File exists


In [ ]:
!macs2 callpeak --broad -t bowtie2_res/res_NCG.sam \
    -c bowtie2_res/res_NCG.sam \
 	  -f SAM \
	  --outdir macs2

INFO  @ Mon, 7 Mar 2022 22:50:26: 
# Command line: callpeak --broad -t bowtie2_res/res_NCG.sam -c bowtie2_res/res_NCG.sam -f SAM --outdir macs2
# ARGUMENTS LIST:
# name = NA
# format = SAM
# ChIP-seq file = ['bowtie2_res/res_NCG.sam']
# control file = ['bowtie2_res/res_NCG.sam']
# effective genome size = 2.70e+09
# band width = 300
# model fold = [5, 50]
# qvalue cutoff for narrow/strong regions = 5.00e-02
# qvalue cutoff for broad/weak regions = 1.00e-01
# The maximum gap between significant sites is assigned as the read length/tag size.
# The minimum length of peaks is assigned as the predicted fragment length "d".
# Larger dataset will be scaled towards smaller dataset.
# Range for calculating regional lambda is: 1000 bps and 10000 bps
# Broad region calling is on
# Paired-End mode is off
 
INFO  @ Mon, 7 Mar 2022 22:50:26: #1 read tag files... 
INFO  @ Mon, 7 Mar 2022 22:50:26: #1 read treatment tags... 
INFO  @ Mon, 7 Mar 2022 22:50:32:  5000000 
INFO  @ Mon, 7 Mar 2022 22:50:36: 

In [ ]:
!macs2 callpeak --broad -t bowtie2_res/res_ERP.sam \
    -c bowtie2_res/res_ERP.sam \
 	  -f SAM \
	  --outdir macs22

INFO  @ Mon, 7 Mar 2022 22:51:36: 
# Command line: callpeak --broad -t bowtie2_res/res_ERP.sam -c bowtie2_res/res_ERP.sam -f SAM --outdir macs22
# ARGUMENTS LIST:
# name = NA
# format = SAM
# ChIP-seq file = ['bowtie2_res/res_ERP.sam']
# control file = ['bowtie2_res/res_ERP.sam']
# effective genome size = 2.70e+09
# band width = 300
# model fold = [5, 50]
# qvalue cutoff for narrow/strong regions = 5.00e-02
# qvalue cutoff for broad/weak regions = 1.00e-01
# The maximum gap between significant sites is assigned as the read length/tag size.
# The minimum length of peaks is assigned as the predicted fragment length "d".
# Larger dataset will be scaled towards smaller dataset.
# Range for calculating regional lambda is: 1000 bps and 10000 bps
# Broad region calling is on
# Paired-End mode is off
 
INFO  @ Mon, 7 Mar 2022 22:51:36: #1 read tag files... 
INFO  @ Mon, 7 Mar 2022 22:51:36: #1 read treatment tags... 
INFO  @ Mon, 7 Mar 2022 22:51:41:  5000000 
INFO  @ Mon, 7 Mar 2022 22:51:46:

In [ ]:
!wget https://www.encodeproject.org/files/ENCFF151QZZ/@@download/ENCFF151QZZ.bed.gz
!gzip -d ENCFF151QZZ.bed.gz

--2022-03-07 22:57:37--  https://www.encodeproject.org/files/ENCFF151QZZ/@@download/ENCFF151QZZ.bed.gz
Resolving www.encodeproject.org (www.encodeproject.org)... 34.211.244.144
Connecting to www.encodeproject.org (www.encodeproject.org)|34.211.244.144|:443... connected.
HTTP request sent, awaiting response... 307 Temporary Redirect
Location: https://encode-public.s3.amazonaws.com/2017/02/01/cc32e3a2-58ff-47fd-b40f-3374a79274b4/ENCFF151QZZ.bed.gz?response-content-disposition=attachment%3B%20filename%3DENCFF151QZZ.bed.gz&AWSAccessKeyId=ASIATGZNGCNX3NA2O5HM&Signature=dko7HgQzfef4NPF3KPaCg%2FLuVeU%3D&x-amz-security-token=IQoJb3JpZ2luX2VjEE8aCXVzLXdlc3QtMiJHMEUCICyaQEmda7%2FZDW5kv2LViJ0%2B1ShyD5VkragvwW3U2dTwAiEAmy6v5bo90miTlt3%2FpclUJNeGgGaxiMxNF3C52iYhTagqgwQI%2BP%2F%2F%2F%2F%2F%2F%2F%2F%2F%2FARAAGgwyMjA3NDg3MTQ4NjMiDDlnhYCtCDKfyE0vkyrXA%2FI0uBTsje1EIxzWLPgt5IZx5qIYWj%2BbYtGGw0Z1k2seXlhDzs%2BCNXv%2BJ%2Bma5L0DAw6vy40guE4GujepUCu1rZm7gHFAGx8i9gvaIaiLPfJXaEYXXHfp4mO%2BX8IFdKoCBA9lUooCNKFseDP

## Сравнение результатов

Сравниваем те пики, которые мы получили, с пиками, которые приведены в ENCODE (важно, чтобы версии генома hg38 или hg19) для .bed файла из ENCODE и той хромосомы, которую скачивали выше, сопадали.

Проанализируйте полученные результаты и приведите свои рассуждения в README.md. Как можно объяснить различия в количестве пересечений?

In [ ]:
!intervene venn -i macs2/NA_peaks.broadPeak ENCFF151QZZ.bed --filenames


Generating a 2-way "venn" diagram. Please wait...


Done! Please check your results @ /content/Intervene_results. 
Thank you for using Intervene!



In [ ]:
!intervene venn -i ENCFF151QZZ.bed macs2/NA_peaks.broadPeak --filenames


Generating a 2-way "venn" diagram. Please wait...


Done! Please check your results @ /content/Intervene_results. 
Thank you for using Intervene!



In [ ]:
!intervene venn -i macs22/NA_peaks.broadPeak ENCFF151QZZ.bed --filenames


Generating a 2-way "venn" diagram. Please wait...


Done! Please check your results @ /content/Intervene_results. 
Thank you for using Intervene!



In [ ]:
!intervene venn -i ENCFF151QZZ.bed macs22/NA_peaks.broadPeak --filenames


Generating a 2-way "venn" diagram. Please wait...


Done! Please check your results @ /content/Intervene_results. 
Thank you for using Intervene!

